test/box-luatest/gh_5529_sort_order_test.lua

```diff
@@ -0,0 +1,180 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g1 = t.group('sort_order_test', {
+    {engine = 'memtx', is_unique = true},
+    {engine = 'memtx', is_unique = false},
+    {engine = 'vinyl', is_unique = true},
+    {engine = 'vinyl', is_unique = false}
+})
```

Comment:
The test should be moved to `engine-luatest` since it's multi-engine.

Also, please add more test cases:
- Errors raised when the sort order is unsupported.
- Snapshot and recovery.
- Reversing the sort order while altering a space.

Reply:
Done.